cli/sawtooth_cli/batch.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2016 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+import os
+import sys
+import csv
+import json
+import yaml
+
+from sawtooth_cli.rest_client import RestClient
+from sawtooth_cli.exceptions import CliException
+
+
+def add_batch_parser(subparsers, parent_parser):
+    parser = subparsers.add_parser('batch')
+
+    grand_parsers = parser.add_subparsers(title='grandchildcommands',
+                                          dest='subcommand')
+    grand_parsers.required = True
+    epilog = '''details:
+        Lists committed batches from newest to oldest, including their id (i.e.
+    header signature), transaction count, and their signer's public key.
+    '''
+
+    list_parser = grand_parsers.add_parser('list', epilog=epilog)
+    list_parser.add_argument(
+        '--url',
+        type=str,
+        help="the URL of the validator's REST API")
+    list_parser.add_argument(
+        '-f', '--format',
```

Comment:
'-f' is commonly used to mean input 'file' or 'force' on many Linux commands, so avoiding using it for other purposes is usually desirable.  Could use '-F', assuming we want '-F' to mean format across the CLI and not something else (we also have --filter in some places).